src/Annotation/Event.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenClassrooms\ServiceProxy\Annotation;
+
+use OpenClassrooms\ServiceProxy\Contract\EventHandler;
+
+/**
+ * @Annotation
+ * @Target({"METHOD"})
+ */
+final class Event extends Annotation
+{
+    public const DEFAULT_METHOD = self::POST_METHOD;
+
+    public const DEFAULT_PREFIX = 'use_case';
+
+    public const ON_EXCEPTION_METHOD = 'onException';
+
+    public const POST_METHOD = 'post';
+
+    public const PRE_METHOD = 'pre';
+
+    /**
+     * @var string[]
+     */
+    private static array $allowedMethods = [
+        self::PRE_METHOD,
+        self::POST_METHOD,
+        self::ON_EXCEPTION_METHOD,
+    ];
+
+    private ?string $defaultPrefix = self::DEFAULT_PREFIX;
+
+    /**
+     * @var array<int, 'pre'|'post'|'onException'>
+     */
+    private array $methods = [self::DEFAULT_METHOD];
+
+    private ?string $name = null;
+
+    private bool $useClassNameOnly = true;
+
+    public function getDefaultPrefix(): string
+    {
+        return $this->defaultPrefix ?: self::DEFAULT_PREFIX;
+    }
+
+    /**
+     * @return array<int, 'pre'|'post'|'onException'>
+     */
+    public function getMethods(): array
+    {
+        return $this->methods;
+    }
+
+    public function getName(): ?string
+    {
+        return $this->name;
+    }
+
+    public function hasMethod(string $method): bool
+    {
+        return \in_array($method, $this->methods, true);
```

Comment:
Maybe can we apply `\` on all PHP functions. What do you think ?

Reply:
Done, added `NativeFunctionInvocationFixer` rule